vignettes/site_only/db_connections.Rmd

```diff
@@ -0,0 +1,183 @@
+---
+title: "Database Connections using healthcare.ai"
+author: "Mike Mastanduno"
+date: "4/1/2018"
+output: html_document
+---
+
+**Database connections**
+
+Let's be honest. You need to get data from a database, do some really awesome 
+stuff with `R`, and then write the results back to a database. `R` can help
+with that. There are lots of different databases, but this document will focus
+on Microsoft SQL Server. R supports most databases, and 
+I'd encourage you to look at their [help pages](https://db.rstudio.com/) 
+for additional resources. Andiamo!
+
+If you simply want a code snippet to copy and paste, scroll to the bottom.
+
+# Reading data
+
+#### Connecting to a Database
+
+Before you can do anything, you must make a connection to your database. The 
+`build_connection_string` function takes care of the syntax for you and returns
+a connection string that's ready to be passed into `DBI::dbConnect`. After
+you have a valid connection, you should be able to browse the database using 
+RStudio's Connections pane. If you don't have a connections pane, upgrade your 
+Rstudio. It's worth it.
+
+```{r, eval=FALSE}
+library(healthcareai)
+library(DBI)
+
+my_con <- build_connection_string(server = "HCS-GM0004",
+                                  database = "SAM")
+con <- DBI::dbConnect(odbc::odbc(), .connection_string = my_con)
+```
+
+
+#### Using SQL Code
+
+From here, you can read data from the database in a couple different ways. You
+can use `db_read` to execute SQL code, or you can use `dbplyr` functionaliity
+to execute `dplyr` code directly against the database.
+```{r, eval=FALSE}
+query <- "SELECT
+          ,COUNT(*),
+          ,year(AdmitDT) as AdmitYear
+          FROM [SAM].[Encounter].[PatientEncounterBASE]
+          WHERE AdmitYear >= 2013
+          GROUP BY AdmitYear
+          ORDER BY AdmitYear"
+
+d <- db_read(con, query)
+```
+This query counts admits by year. If you don't want to pull the results into
+memory, as might the be the case with large tables, set the `pull_into_memory`
+flag to FALSE. This will create a pointer to the database that can be 
+executed "lazily." In other words R will wait until the last moment to execute 
+the statement.
+
+#### Using dplyr Code
+
+The other option is to use dplyr code. Set up a reference to the table, then use
```

Comment:
I'd be more explicit here that `tbl` is creating the reference, which gets stored in `encounter` and what you can do with that... some but not all dplyr code to manipulate the data without pulling it all into memory

Reply:
added some notes about this.